slo/operators.py

```diff
@@ -387,12 +396,17 @@ def _eval(self, y, x, alpha=1, beta=0, forward=True):
         batch = x.shape[1]
         tmp = self._get_or_create_intermediate( batch, x.dtype )
         L, R = self._children
-        if forward:
-            R.eval(tmp, x, alpha=alpha, beta=0, forward=True)
-            L.eval(y, tmp, alpha=1,  beta=beta, forward=True)
-        else:
-            L.eval(tmp, x, alpha=alpha, beta=0, forward=False)
-            R.eval(y, tmp, alpha=1,  beta=beta, forward=False)
+        bs = self._batch_size if self._batch_size != None else batch
```

Comment:
cool python trick, but with slightly different semantics so feel free to ignore: `bs = self._batch_size or batch`